lib/datadog/tracing/configuration/settings.rb

```diff
@@ -51,14 +53,13 @@ def self.extended(base)
                 #   otherwise `['Datadog','b3multi','b3']`.
                 # @return [Array<String>]
                 option :propagation_extract_style do |o|
-                  o.default do
+                  o.deprecated_env_var Tracing::Configuration::Ext::Distributed::ENV_PROPAGATION_STYLE_EXTRACT_OLD
```

Comment:
I may have counted them wrong, but it seems to me we're adding `deprecated_env_var` to support only 2 configuration settings -- this one and `ENV_PROPAGATION_STYLE_INJECT_OLD`.

It seems a lot of work for only these two use-cases... Could we perhaps use custom logic on the `setter` or somewhere else to avoid having the configuration system need to support this?